### ISSUE TITLE
permission denied

### ISSUE THREAD

**Issue by auslegungssache** [bhgv/Inferno-OS_Android#6]
Whenever i try to modify a file in the app, i get an error like...
> mkdir: can't create /n/client: permission denied

Is that correct behaviour? How can i fix this? Thanks in advance

**Comment by bhgv** [comment]
hmm, do you have '/n' folder in /sdcard/inferno?

run please in your just loaded inferno in the shell:
```
ls /n
```

does it show /n? or it shows nothing? if so - 
mkdir /n

**Comment by auslegungssache** [comment]
That's not the problem, i can solve that with mntgen. The problem is that i can't modify files/directories at all.

**Comment by eekee** [comment]
how do you "solve it with mntgen"?  do you mount mntgen on /n or /?
mntgen only works at one level, so if you mount it on / to cause /n to
appear, you'll get 'permission denied' whenever you try to create
anything in /n.
% mkdir q
% mount {mntgen} q
% cd q
% mkdir -p a/b
mkdir: can't create ./a/b: permission denied
%

**Comment by auslegungssache** [comment]
No, i mount it on /n. It was just an example. You can't delete/edit files/directories in the Inferno app at all

**Comment by eekee** [comment]
On Sat, Jan 12, 2019, at 7:49 AM, Adam Kučerňák wrote:
> No, i mount it on /n. It was just an example. You can't delete/edit files/directories in the Inferno app at all

Ah ok.  Maybe it's Android app permissions then; permission to write to storage.  That can get weird because you don't get to approve it at install time any more.  Termux has a program you run to request the permission; I recall being surprised by "permission denied" errors before I ran it, but don't remember the details.

**Comment by auslegungssache** [comment]
Wouldn't it be possible to move the Inferno folder in the app data folder on runtime?

**Comment by bhgv** [comment]
checked your note about '/n' folder just now.

'/n' is a synthetic, not a real folder. as '/dev' folder for example. it is not present in the phisical filesystem. 
and it is not enabled for the part of filesystem operations.
it is only a name on filesystem tree.
but you may mount and bind to it. (do not remember preciselly why i did /n folder virtual. maybe for testing?)

for example you may mount to /n a memory filesystem to do filesystem operations on it.
or you may 'mkdir /m' (or '/nn') and do what you want on '/m' (or on '/nn').
or remove defining of /n as a virtual from inferno's source (near 'main'). search in the INFERNO_ROOT/emu folder by the word "/n". and recompile
